api/api.py

```diff
@@ -174,6 +176,45 @@ def post(self):
         return {}, 200
 
 
+class APDU:
+    def __init__(self):
+        # We want to be notified when APDU response is transmitted from the SE
+        self.condition = threading.Condition()
+
+    def exchange(self, data: bytes) -> bytes:
+        with self.condition:  # Lock for a command/response for one client
+            self.response = None
+            seph.to_app(data)
+            with self.condition:
```

Comment:
Why is the function locking the same condition twice?

Reply:
This is a mistake, I think first lock should be a separate `threading.Lock` since `wait` releases `condition` until response is available. Would it be ok for you ?

Reply:
What is the semantics of `self.condition`? There seems to be 2 possible meanings:
1. a lock on `self.response`, associated with signaling so that `exchange` can wait for `seph_apdu_callback` to receive response data.
2. a lock on the `/apdu` endpoint so that only one client can use it.

If the second meaning was expected, the current code is buggy, because the lock is released when `exchange` waits for the response data. A dedicated lock is then needed, for this second meaning.

Otherwise if the second meaning is not expected (is it possible to send another command before a response was received?), the comment `Lock for a command/response for one client` should be removed.

Reply:
Both 1 and 2 are expected. `self.condition` should satisfy response waiting, and another `self.lock` should satisfy endpoint locking. I can commit a change if you do agree? (actually I had those two locks at the beginning and thought by mistake I could use only one).

Reply:
Fixed. Thank you niooss for helping fixing this part.